### PR TITLE
Updating rendering script 

### PR DESCRIPTION
**Pull request by vcharpenay**
Rendering is now more closely aligned with the SHACL schema language and several mistakes in `validation/td-validation.ttl` have been fixed. Documentation of rendering is available as comments in `render.sh`.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/951.html" title="Last updated on Sep 19, 2020, 12:13 AM UTC (93c074f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/951/32ba69e...93c074f.html" title="Last updated on Sep 19, 2020, 12:13 AM UTC (93c074f)">Diff</a>

### CODE REVIEW

**Comment by danielpeintner** [comment]
@vcharpenay thank you for your work

> Documentation of rendering is available as comments in `render.sh`.

Great. I was able to modify `validation/td-validation.ttl`  and adding minLength to StringArraySchema. Everything seems to work as expected.

Some more comments (you might be aware of I guess):

### index.template.html
* switch the PNG image versions to SVG again

### td-validation.ttl 
* check "at risk" features and TODO comments for security (can be mostly removed I think)
* How to deal with all the manual tweaking that has been done in the security sections (needs a lot of caution I think)
* is there a good tool how to properly validate ttl changes. I was not very succesful with https://shacl.org/playground/

File: render.sh [hunk 1]
```diff
@@ -1,24 +1,132 @@
 #!/bin/bash
 
-if ! [[ -e .install/fuseki.jar ]] ; then
-	mkdir .install
-	echo "Downloading RDF store (Apache Fuseki server)..."
-	curl https://repo1.maven.org/maven2/org/apache/jena/jena-fuseki-server/3.9.0/jena-fuseki-server-3.9.0.jar -o .install/fuseki.jar
+# The parts of the TD specification encoded in RDF are rendered using STTL,
```
Comment (danielpeintner) [inline render.sh:3]: Maybe we can add a short requirements statement before that to specify the needs..

### Requirements:
* Java
* graphviz  (sudo apt install graphviz )

**Comment by takuki** [comment]
In TD call on Sept 2, it was suggested to merge this with "index2.html" instead of index.html.
We can retire index2.html after all conflicts are resolved with index.html.
Same for index.template.html. It should be merged into "index2.template.html" for now.

For wot-security.ttl, it should be re-based before merging this PR.

**Comment by vcharpenay** [comment]
i resolved conflicts in the various files. The single origin of conflicts were changes on `OAuth2SecurityScheme`: @mmccool please, review. I changed a bit the ordering of the HTML paragraphs, let me know if it was fit. However, I think creating subclasses of the OAuth2 scheme (one per flow) would greatly simplify the spec. Let me know if you need help.

> it was suggested to merge this with "index2.html" instead of index.html.
We can retire index2.html after all conflicts are resolved with index.html.

Now that I've committed changes to `index.html`, I'm not sure I understand what I was supposed to do... All changes in the rendered part, in `index.html` related to OAuth2 or were also added to `index.template.html`. There shouldn't be any lost change (but it is always possible to revert). Could you please review @takuki?

Note: no rebase is needed for `wot-security.ttl`: changes were to be moved elsewhere, the file must simply be deleted.

**Comment by sebastiankb** [comment]
from today's TD call:
- @mmccool will review this PR until tomorrow
- if the review is ok from @mmccool then he will merge the PR

**Comment by mmccool** [comment]
Sorry this took so long.  What matters to me is whether the generated index.html is the same as what it was before or not.  To check that I made a [diff](
https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fwot-thing-description%2Fmaster%2Findex.html&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fwot-thing-description%2Ffix-render%2Findex.html).

**Review by mmccool — APPROVED** [review]
I made a diff of the new index.html file and the changes are acceptable, although there are three old schemes that seem to have gotten (accidentally) resurrected (Cert, Public, and PoP).   In the interest of time I'm going to merge this PR anyway but will create an issue to review these.

**Comment by mmccool** [comment]
As discussed, merging.   Will create an issue to deal with resurrected schemes (Cert, Public, and PoP).

**Comment by mmccool** [comment]
BTW I agree making subclasses for each OAuth2 flow would be desirable.   Let me create an issue for that, too.

**Pull request merged**